lib/harmonyclient.js

```diff
@@ -17,6 +17,7 @@ function HarmonyClient (xmppClient) {
 
   self._xmppClient = xmppClient
   self._responseHandlerQueue = []
+  self._cachedConfig = null
```

Comment:
the value we cache would not really be config. what about calling it `_availableCommandsCache` instead?

```suggestion
  self._availableCommandsCache = null
```

Reply:
I'm going to make the change, because it's your project, and I have no reason to fight with you over it. But...

I'm not sure I agree with this statement.  I am _literally_ caching the response from the `config` request. How is it not caching the config?

I think it makes more sense to cache the `config` response, and then if you need to manipulate it in some way (like extracting only the "available commands" out of that config), that can still be done on the cached object (or a freshly fetched object, doesn't matter).

Caching this as `availableCommandsCache` would make sense if the cached object were some kind of manipulation of the data fetched from Harmony, but it's not. It's just literally the entire Harmony config.

Reply:
Renamed to `_availableCommandsCache`, and addressed the build failure for code styles.

I think the trap that this is falling into now, is that "available commands" is a really bad definition of what is being cached here.  There's a lot more in that configuration than just the commands, and the remote endpoint is actually called `config` (for a reason), and not just `availablecommands`.

For example, if `getAvailableCommands()` were to manipulate the config to return only the top level of commands, then the cache of just that manipulation is not very effective at all.  And instead it would make more sense to rename this method to `getConfig()`, cache it as `configCache`, and make a _new_ method for `getAvailableCommands()` that utilizes `getConfig()`, extracting whatever it needs in order to accomplish that.

But the new patch follows your suggestion, so any of the other concepts mentioned above are just theoretical anyway, at least for now.